3rdparty/llvm/CMakeLists.txt

```diff
@@ -17,7 +17,8 @@ if(WITH_LLVM)
 		option(LLVM_CCACHE_BUILD OFF)
 		set(LLVM_ENABLE_WARNINGS OFF CACHE BOOL "Enable compiler warnings.")
 
-		if(WIN32 AND COMPILER_X86)
+		# For non arm64 windows
+		if(WIN32 AND (COMPILER_X86 OR (MSVC AND NOT COMPILER_ARM)))
```

Comment:
MSVC-amd64 should set COMPILER_X86. Is this not happening right now?

Reply:
It is better to change the lines 73-78 to
```
		if(LLVM_USE_INTEL_JITEVENTS)
			list (APPEND LLVM_ADDITIONAL_LIBS IntelJITEvents)
		endif()
		if(LLVM_USE_PERF)
			list (APPEND LLVM_ADDITIONAL_LIBS PerfJITEvents)
		endif()
```
I tested it and it worked.

Reply:
MSVC for arm isn't different than the one for X64 so it won't set COMPILER_ARM either.
The condition should just be `if(WIN32 AND (COMPILER_X86 OR MSVC))`. We don't support using msvc for arm64 windows building due to extensive problems with microsoft's compiler stack.

Reply:
yes, if COMPILER_X64 is not set on MSVC there are also few chances COMPILER_ARM will be set for that arch.

Reply:
You should apply my changes in addition to that. If you look at the llvm cmake code you will see that `IntelJITEvents` is only built if `LLVM_USE_INTEL_JITEVENTS` is true and `PerfJITEvents` is only built if `LLVM_USE_PERF` is true.

Reply:
> You should apply my changes in addition to that. If you look at the llvm cmake code you will see that `IntelJITEvents` is only built if `LLVM_USE_INTEL_JITEVENTS` is true and `PerfJITEvents` is only built if `LLVM_USE_PERF` is true.

Yes, I reported that in the other post in this thread. I already integrated your changes and kd-11 changes. Just making other checks and tests before I will push the changes

Reply:
fixed

Reply:
at the end the code needed for building on all supported OS and architecture was:

`if(WIN32 AND (COMPILER_X86 OR MSVC))`

and

```
if(LLVM_USE_INTEL_JITEVENTS OR (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND COMPILER_X86))
	list (APPEND LLVM_ADDITIONAL_LIBS IntelJITEvents)
endif()
if(LLVM_USE_PERF OR CMAKE_SYSTEM_NAME STREQUAL "Linux")
	list (APPEND LLVM_ADDITIONAL_LIBS PerfJITEvents)
endif()

```

The check on `Linux` is needed for `gcc` and `clang`  due to the following options set in `CMakePresets.json`:

```
"BUILD_LLVM": "OFF",
"STATIC_LINK_LLVM": "ON"

```

With those options and without the check on `Linux`, libs `IntelJITEvents` (in particualr) and `PerfJITEvents` were not linked causing link errors such as the following:

```
ld.lld: error: undefined symbol: llvm::JITEventListener::createIntelJITEventListener()
>>> referenced by JITLLVM.cpp

```